debian/mariadb-server.mariadb.init

```diff
@@ -156,19 +176,41 @@ case "${1:-''}" in
     # Start MariaDB!
     /usr/bin/mariadbd-safe "${@:2}" 2>&1 >/dev/null | $ERR_LOGGER &
 
-    for i in $(seq 1 "${MYSQLD_STARTUP_TIMEOUT:-30}"); do
+    for _ in $(seq 1 "${MYSQLD_STARTUP_TIMEOUT:-30}")
+    do
       sleep 1
-      if mariadbd_status check_alive nowarn ; then break; fi
+      if mariadbd_status check_alive nowarn
+      then
+        break
+      fi
       log_progress_msg "."
     done
-    if mariadbd_status check_alive warn; then
+    if mariadbd_status check_alive warn
+    then
       log_end_msg 0
       # Now start mysqlcheck or whatever the admin wants.
       output=$(/etc/mysql/debian-start)
-      if [ -n "$output" ]; then
+      if [ -n "$output" ]
+      then
         log_action_msg "$output"
       fi
     else
+      # Try one more time but save error log separately, then spit it out
+      # before logging ends and init script execution ends.
+      if pgrep -ax mariadbd > /dev/null
+      then
+        echo "ERROR: The mariadbd process is running but not responding:"
+        # shellcheck disable=SC2009
+        # Show the mariadbd process and it's parent and next line (if there is a child process)
+        ps faxu | grep mariadbd -C 1
```

Comment:
`ps -C mariadbd -o ppid` or something?

Reply:
I'm not exactly sure what is intend to achieve in this one. Would `pgrep` give wanted as it gives parent pid and child pid?

Reply:
it assumes the subprocess it somehow next. werid. sysint stuff so I don't particularly care.